src/pages/SignIn.js

```diff
@@ -0,0 +1,152 @@
+
+export default function SignIn() {
+    const [email, setEmail] = useState("");
+    const [password, setPassword] = useState("");
+    let navigate = useNavigate();
+
+    useEffect(() => {
+        // Get isLoggedIn from Session Storage
+        let isLoggedIn = sessionStorage.getItem('isLoggedIn');
+
+        // If isLoggedIn is true, redirect to /dashboard
+        if (isLoggedIn === 'true') {
+            navigate('/dashboard');
+        }
+    }, [navigate]);
+
+    const handleSubmit = (event) => {
+        event.preventDefault();
+        const isLoggedIn = sessionStorage.getItem('isLoggedIn');
+        if (isLoggedIn === "") {
+            alert("user already logged in.")
+            return
+        }
+        const users = JSON.parse(localStorage.getItem('userDetails')) || [];
+
+        const user = users.find(user => user.email === email);
+        if (user) {
+            var pass = (CryptoJS.AES.decrypt(user.password, 'secret_key_here_lol').toString(CryptoJS.enc.Utf8));
+            if (pass === password) {
+                sessionStorage.setItem('isLoggedIn', true);
+                sessionStorage.setItem('loggedUser', user.email);
+                window.location.href = "/dashboard";
+            }
+            else {
+                alert("Wrong Password!");
+            }
+        }
+        else {
+            alert("No User Found with provided Email ID.");
+        }
+    };
+
+    const [showPassword, setShowPassword] = useState(false);
+    const handleTogglePasswordVisibility = () => setShowPassword(!showPassword);
+    const handleMouseDownPassword = () => setShowPassword(!showPassword);
+
+    return (
+        <ThemeProvider theme={defaultTheme}>
+            <Container component="main" maxWidth="xs">
+                <CssBaseline />
+                <Box
+                    sx={{
+                        marginTop: 8,
+                        display: 'flex',
+                        flexDirection: 'column',
+                        alignItems: 'center',
+                    }}
+                >
+                    <Avatar sx={{ m: 1, bgcolor: 'secondary.main' }}>
+                        <LockOutlinedIcon />
+                    </Avatar>
+                    <Typography component="h1" variant="h5">
+                        Sign in
+                    </Typography>
+                    <Box component="form" onSubmit={handleSubmit} sx={{ mt: 1 }}>
+                        {/* Email address */}
+
+                        <TextField
+                            required
+                            fullWidth
+                            id="email"
+                            label="Email Address"
+                            name="email"
+                            autoComplete="email"
+                            value={email}
+                            helperText="Valid hawk or gmail id"
+                            inputProps={{ pattern: "^[a-zA-Z0-9]+@(hawk.iit.edu|gmail.com)$" }}
```

Comment:
Allow "." in the pattern match. [I got error on my email: anusha.sonte@gmail.com]

Reply:
Done